src/k2/transport/TXEndpoint.h

```diff
@@ -44,7 +44,7 @@ class TXEndpoint {
 public: // lifecycle
     // construct an endpoint from a url with the given allocator
     // Returns nullptr if there was a problem parsing the url
-    static std::unique_ptr<TXEndpoint> fromURL(const String& url, BinaryAllocatorFunctor&& allocator);
+    static std::unique_ptr<TXEndpoint> fromURL(const String& url, BinaryAllocator allocator);
```

Comment:
why this change? you should still receive by && to avoid a copy

Reply:
done